include/aws/http/private/http_impl.h

```diff
@@ -49,9 +61,15 @@ AWS_HTTP_API void aws_http_fatal_assert_library_initialized(void);
 
 AWS_HTTP_API struct aws_byte_cursor aws_http_version_to_str(enum aws_http_version version);
 
+/* case-sensitive */
 AWS_HTTP_API enum aws_http_method aws_http_str_to_method(struct aws_byte_cursor cursor);
+
+/* case-insensitive */
 AWS_HTTP_API enum aws_http_header_name aws_http_str_to_header_name(struct aws_byte_cursor cursor);
 
+/* case-sensitive */
+AWS_HTTP_API enum aws_http_header_name aws_http_lowercase_str_to_header_name(struct aws_byte_cursor cursor);
+
```

Comment:
I'm confused how this behaves with headers that aren't explicitly in the enum, could you document?